main.cpp

```diff
@@ -602,8 +602,9 @@ static bool add_endpoints(Mainloop &mainloop)
             return false;
         }
 
-        g_endpoints[i++] = udp.release();
-        mainloop.add_fd(udp->fd, g_endpoints[i], EPOLLIN);
+        g_endpoints[i] = udp.release();
```

Comment:
Hmmm... about `release`, I actually tested that - if I use `udp->fd` I still get segfault.